internal/querycoord/segment_allocator.go

```diff
@@ -99,13 +100,24 @@ func shuffleSegmentsToQueryNodeV2(ctx context.Context, reqs []*querypb.LoadSegme
 		return nil
 	}
 	log.Debug("shuffleSegmentsToQueryNodeV2: start estimate the size of loadReqs")
-	dataSizePerReq := make([]int64, 0)
-	for _, req := range reqs {
-		sizeOfReq, err := cluster.estimateSegmentsSize(req)
+	dataSizePerReq := make([]int64, len(reqs))
+	estimateError := make([]error, len(reqs))
+	var estimateWg sync.WaitGroup
+	estimateReqFn := func(offset int, req *querypb.LoadSegmentsRequest) {
+		defer estimateWg.Done()
+		dataSizePerReq[offset], estimateError[offset] = cluster.estimateSegmentsSize(req)
+	}
+	for offset, req := range reqs {
+		estimateWg.Add(1)
+		tempReq := req
```

Comment:
Copying here is not necessary. This is not a closure catch.

Reply:
fixed